aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -112,12 +112,65 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
     );
   });
 }
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y, bool min) {
+  scalar_t xr = std::real(x);
+  scalar_t yr = std::real(y);
+  if (std::isnan(yr) || (std::isnan(std::imag(y)))) {
+    return y;
+  } else if (std::isnan(xr) || (std::isnan(std::imag(x)))) {
+    return x;
+  } else {
+    return ((xr < yr) == min) ? x : y;  // logical xnor
+  }
+}
+
+template <typename scalar_t>
+scalar_t _log_add_exp_helper(scalar_t x, scalar_t y) {
+  // Reference : https://www.tensorflow.org/api_docs/python/tf/math/cumulative_logsumexp
+  scalar_t min = std::isnan(y) ? y : std::min(x, y); // std::min returns first arg if one of the args is nan
+  scalar_t max = std::isnan(y) ? y : std::max(x, y); // std::max returns first arg if one of the args is nan
+  if (min != max || std::isfinite(min)) {
+    // nan will be propagated here
+    return std::log1p(std::exp(min - max)) + max;
+  } else {
+    // special case to correctly handle infinite cases
+    return x;
+  }
+}
+
+template <typename scalar_t>
+c10::complex<scalar_t> _log_add_exp_helper(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
```

Comment:
Why not pass by reference?
```suggestion
c10::complex<scalar_t> _log_add_exp_helper(const c10::complex<scalar_t>& x, const c10::complex<scalar_t>& y) {
```

Reply:
Done in the next commit